3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala

```diff
@@ -311,52 +311,44 @@ object ConversionUtils {
     URLEncoder.encode(str, enc)
 
   /**
-   * A wrapper around Java's
-   * URI.create().
-   *
-   * Exceptions thrown by
-   * URI.create():
-   * 1. NullPointerException
-   *    if uri is null
-   * 2. IllegalArgumentException
-   *    if uri violates RFC 2396
-   *
-   * @param uri The URI string to
-   *        convert
-   * @param useNetaporter Whether to use the
-   *        com.netaporter.uri library
-   * @return an Option-boxed URI object, or an
-   *         error message, all
-   *         wrapped in a Validation
+   * Creates a [[URI]] from a string.
+   *
+   * A "relaxed" parsing can be done if `relaxed` is set to true.
+   *
+   * Few examples of URIs that would fail non-relaxed parsing:
+   * - URI having at least one % not followed by 2 hexadecimal digits
+   * - URI with more than one #
+   * - URI containing a |
+   * Such URIs would need a relaxed parsing.
+   *
+   * @param uri String containing the URI to parse.
+   * @param relaxed Specifies whether parsing should be relaxed or not.
+   *
+   * @return [[Validation]] wrapping the result of the parsing:
+   *         - [[Success]] with the parsed URI if there was no error or with [[None]] if the input was `null`.
+   *         - [[Failure]] with the error message if something went wrong.
    */
-  def stringToUri(uri: String, useNetaporter: Boolean = false): Validation[String, Option[URI]] =
+  def stringToUri(uri: String, relaxed: Boolean = false): Validation[String, Option[URI]] =
```

Comment:
should we ditch `relaxed` and use scala-uri directly?

Reply:
I hesitated about it, but I think that it should be left as a possibility to do a strict URI parsing. But we can have a relaxed URI parsing by default and make it strict optionally, instead of the other way around. WDYT?

For example see [this](https://github.com/snowplow/snowplow/blob/master/3-enrich/scala-common-enrich/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/enrichments/registry/EnrichmentConfigsSpec.scala#L446)

Reply:
makes sense, in this case we could split the two, one that uses java and is strict and one that uses scala-uri for non-strict usages

Reply:
Actually, the choice of relaxed parsing or not would still need to go all the way from the configuration to the function:
```
$ grep -R stringToUri
enrichments/EnrichmentManager.scala:    val refererUri = CU.stringToUri(event.page_referrer)
enrichments/registry/IpLookupsEnrichment.scala:      .stringToUri(uri + "/" + database)
enrichments/registry/IabEnrichment.scala:      .stringToUri(uri + (if (uri.endsWith("/")) "" else "/") + database)
enrichments/registry/UaParserEnrichment.scala:      .stringToUri(uri + (if (uri.endsWith("/")) "" else "/") + database)
enrichments/web/PageEnrichments.scala:      case (Some(r), None)    => CU.stringToUri(r)
enrichments/web/PageEnrichments.scala:      case (None, Some(t))    => CU.stringToUri(t)
enrichments/web/PageEnrichments.scala:      case (Some(r), Some(t)) => CU.stringToUri(t) // Tracker URL takes precedence
utils/ConversionUtils.scala:  def stringToUri(uri: String, relaxed: Boolean = true): Validation[String, Option[URI]] =
```
Although it's technically possible, I'm not sure that this is worth it to make it happen. Can you think of any case where we wouldn't want a relaxed URI parsing? If not I think that we should just always use relaxed parsing, as you suggested.